src/toil_scripts/lib/files.py

```diff
@@ -0,0 +1,81 @@
+from contextlib import closing
+import os
+import errno
+import tarfile
+import shutil
+
+
+def mkdir_p(path):
+    """
+    It is Easier to Ask for Forgiveness than Permission
+
+    :param str path: path to directory to create
+    """
+    try:
+        os.makedirs(path)
+    except OSError as exc:
+        if exc.errno == errno.EEXIST and os.path.isdir(path):
+            pass
+        else:
+            raise
+
+
+def tarball_files(tar_name, work_dir='.', fpaths=None, uuid=None):
+    """
+    Creates a tarball from a group of files
+
+    :param str tar_name: Name of tarball
+    :param str work_dir: Current working directory
+    :param list fpaths: File paths to include in the tarball
+    :param str uuid: UUID to stamp files with
+    """
+    if fpaths is None:
+        fpaths = []
+    with tarfile.open(os.path.join(work_dir, tar_name), 'w:gz') as f_out:
+        for fpath in fpaths:
+            if uuid:
+                f_out.add(fpath, arcname=uuid + '.' + os.path.basename(fpath))
+            else:
+                f_out.add(fpath, arcname=os.path.basename(fpath))
+
+
+def move_to_output_dir(output_dir, filepaths=None):
+    """
+    Moves files from the working directory to the output directory.
+
+    :param str output_dir: Output directory
+    :param list filepaths: Filepaths to move
+    """
+    if filepaths is None:
+        filepaths = []
+    for fpath in filepaths:
+        dest = os.path.join(output_dir, os.path.basename(fpath))
+        shutil.move(fpath, dest)
+
+
+def consolidate_tarballs_job(job, **kwargs):
+    """
+    Combine the contents of separate tarballs into one.
+    Subdirs within the tarball will be named the keys in **kwargs.
+
+    :param dict kwargs: Key word arguments of the form: file_name_prefix=FileStoreID
+    :return str: FileStoreID
+    """
+    work_dir = job.fileStore.getLocalTempDir()
+    # Retrieve output file paths to consolidate
+    tar_paths = []
+    for fname in kwargs:
+        p = job.fileStore.readGlobalFile(kwargs[fname], os.path.join(work_dir, fname + '.tar.gz'))
+        tar_paths.append((p, fname))
+    # I/O
+    output_name = 'foo.tar.gz'
+    out_tar = os.path.join(work_dir, output_name)
+    # Consolidate separate tarballs into one
+    with tarfile.open(os.path.join(work_dir, out_tar), 'w:gz') as f_out:
+        for tar, fname in tar_paths:
+            with tarfile.open(tar, 'r') as f_in:
+                for tarinfo in f_in:
+                    with closing(f_in.extractfile(tarinfo)) as f_in_file:
+                        tarinfo.name = os.path.join(output_name, fname, os.path.basename(tarinfo.name))
+                        f_out.addfile(tarinfo, fileobj=f_in_file)
+    return job.fileStore.writeGlobalFile(out_tar)
```

Comment:
I wonder if the generality of this method would be increased if you removed the call to wGF().